contracts-periphery/src/UmbraBatchSend.sol

```diff
@@ -0,0 +1,93 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.13;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+interface IUmbra {
+  function sendEth(
+    address payable receiver,
+    uint256 tollCommitment,
+    bytes32 pkx,
+    bytes32 ciphertext
+  ) external payable;
+
+  function sendToken(
+    address receiver,
+    address tokenAddr,
+    uint256 amount,
+    bytes32 pkx,
+    bytes32 ciphertext
+  ) external payable;
+}
+
+contract UmbraBatchSend {
+  IUmbra internal constant umbra = IUmbra(0xFb2dc580Eed955B528407b4d36FfaFe3da685401);
+
+  struct SendEth {
+    address payable receiver;
+    uint256 amount;
+    bytes32 pkx;
+    bytes32 ciphertext;
+  }
+
+  struct SendToken {
+    address receiver;
+    address tokenAddr;
+    uint256 amount;
+    bytes32 pkx;
+    bytes32 ciphertext;
+  }
+  
+  error ValueMismatch();
+  event Log(address indexed caller, uint256 indexed value, string message);
+
+  function batchSendEth(uint256 _tollCommitment, SendEth[] calldata _params) public payable {
+    uint256 valAccumulator;
+
+    for (uint256 i = 0; i < _params.length; i++) {
+      //amount to be sent per receiver
+      uint256 _amount = _params[i].amount;
+      valAccumulator += _amount;
+
+      umbra.sendEth{value: _amount}(_params[i].receiver, _tollCommitment, _params[i].pkx, _params[i].ciphertext);
+    }
+
+    if(msg.value != valAccumulator) revert ValueMismatch();
+    emit Log(msg.sender, msg.value, "called batchSendEth");
+  }
+
+  function batchSendTokens(uint256 _tollCommitment, SendToken[] calldata _params) public payable {
```

Comment:
This method needs a check that `_tollCommitment * _params.length == msg.value` to prevent excess ETH from getting stuck

Reply:
Looks like this was added but no test was added for it. In general there should be at least 1 test for each condition checked in a function that could cause a revert (currently I don't see any `expectRevert` tests)